src/main/scala/play/dev/filewatch/PollingFileWatchService.scala

```diff
@@ -41,13 +41,21 @@ class PollingFileWatchService(val pollDelayMillis: Int) extends FileWatchService
 object SourceModificationWatch {
   type PathFinder = () => Iterator[ScalaFile]
 
+  private def listFiles(sourcesFinder: PathFinder): Set[ScalaFile] = sourcesFinder().toSet
+
+  private def findLastModifiedTime(files: Set[ScalaFile]): Long = {
+    if (files.nonEmpty) files.maxBy(_.lastModifiedTime).lastModifiedTime.toEpochMilli
```

Comment:
There should be not much performance difference between using `maxBy` instead of `foldLeft`, but I thought this would make the code clearer.